File: aScharfe/ArabicRomanKata/ArabicRomanKata/Helper/CheckInputValue.cs
using System;
using System.Text.RegularExpressions;
using ArabicRomanKata.Enum;

namespace ArabicRomanKata.Helper
{
    /// <summary>
    /// 
    /// </summary>
    public static class CheckInputValue
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="inputValue"></param>
        /// <returns></returns>
        public static bool CheckValueIsValid(this string inputValue)
        {
            if (string.IsNullOrEmpty(inputValue))
            {
                Console.WriteLine("No value was specified for the conversion.");
                return false;
            }

            if (int.TryParse(inputValue, out var testValue))
            {
                if (testValue > 0 & testValue < 4000)
                {
                    return true;
                }

                Console.WriteLine("\"{0}\t\"The Arabic numeral is outside the valid range. Valid are numbers from 1 - 3999.", inputValue);
                return false;
            }

            var successfulRomanMatch = Regex.Match(inputValue, Global.RomanNumberRegExExpression).Success;

            if (!successfulRomanMatch)
                Console.WriteLine("\"{0}\"\tIs not a valid Roman numeral and not a valid Arabic numeral.", inputValue);
            return successfulRomanMatch;
        }

        public static ConversionDirection GetConversionDirection(this string inputValue)
        {
            if (string.IsNullOrEmpty(inputValue)) return ConversionDirection.Unknown;
            if (Regex.Match(inputValue, Global.RomanNumberRegExExpression, RegexOptions.IgnoreCase).Success)
                return ConversionDirection.ToArabic;
            if (int.TryParse(inputValue, out _))
                return ConversionDirection.ToRoman;

            return ConversionDirection.Unknown;
        }
    }


}

File: aScharfe/ArabicRomanKata_Python_aScharfe/ArabicRomanKata_Python/Helper/CheckInputValue.py
import re

class CheckInputValue():
    
    def  CheckValueIsValid(inputValue):
        if CheckInputValue.RepresentsInt(inputValue):
            arabicNumber = int(inputValue)
            if (arabicNumber > 0) & (arabicNumber < 4000):
                return True
            else:
                print(inputValue + "    The Arabic numeral is outside the valid range. Valid are numbers from 1 - 3999.")
                return False
        elif CheckInputValue.RepresentsRomanNumber(inputValue):
            return True
        else:
            print(inputValue + " Is not a valid Roman numeral and not a valid Arabic numeral.")
            return False

    def GetConversionDirection(inputValue):
        if CheckInputValue.RepresentsRomanNumber(inputValue):
            return 1
        if CheckInputValue.RepresentsInt(inputValue):
            return 2

        return 0

    def RepresentsInt(stringValue):
        try:
            int(stringValue)
            return True
        except ValueError:
            return False

    def RepresentsRomanNumber(stringValue):
        if (len(stringValue) == 0):
            return False

        match = re.match("^M{0,4}(CM|CD|D?C{0,3})(XC|XL|L?X{0,3})(IX|IV|V?I{0,3})$",stringValue,re.IGNORECASE)
        if (match != None):
            return True
        else:
            return False
